### PR TITLE
fix: fixes react UI issues which prevented host creation and edit

### PR DESCRIPTION
**Pull request by Manisha15**

### CODE REVIEW

File: app/models/foreman_fog_proxmox/proxmox_interfaces.rb [hunk 1]
```diff
@@ -118,7 +118,7 @@ def dhcp?(nic_compute_attributes, v6 = false)
     def set_mac(nic_compute_attributes, mac, type)
       mac_attr_name = { 'qemu' => :macaddr, 'lxc' => :hwaddr }
       mac_key = mac_attr_name[type] || 'mac'
-      nic_compute_attributes[mac_key] = Net::Validations.normalize_mac(mac)
```
Comment (sbernhard) [inline app/models/foreman_fog_proxmox/proxmox_interfaces.rb:121]: This method does more like only downcase the MAC address. See: https://github.com/theforeman/foreman/blob/develop/lib/net/validations.rb#L122

would it be better to 1) run normalize_mac and afterwards do a upcase (if promox requires the mac address to be upcase?)

File: webpack/components/GeneralTabContent.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ const GeneralTabContent = ({
       required
       type="number"
       value={general?.vmid?.value}
-      disabled={fromProfile}
+      disabled={!newVm || fromProfile}
```
Comment (sbernhard) [inline webpack/components/GeneralTabContent.js:25]: for better tracking, we should have own PRs to fix different issues and to have commit message what the purpose of the PR / commit is.

File: app/helpers/proxmox_vm_attrs_helper.rb [hunk 3]
```diff
@@ -114,11 +114,11 @@ def additional_attrs(vms, param_scope, start_checked)
       start_after_create: vms.start_after_create,
       templated: vms.templated,
     }
-    vms_keys = [:pool, :start_after_create]
+    vms_keys = [:cpu_type, :nameserver, :searchdomain, :hostname]
     extra_attrs = ActiveSupport::HashWithIndifferentAccess.new
     attributes.each do |key, value|
       camel_key = key.to_s.include?('_') ? snake_to_camel(key.to_s).to_sym : key
-      nested_key = vms_keys.include?(key) ? key : "config_attributes[#{key}]"
+      nested_key = vms_keys.include?(key) ? "config_attributes[#{key}]" : key
```
Comment (sbernhard) [inline app/helpers/proxmox_vm_attrs_helper.rb:121]: What is the purpose of this?

File: app/models/foreman_fog_proxmox/proxmox_interfaces.rb (hunk 1 above)
Comment (Manisha15) [reply app/models/foreman_fog_proxmox/proxmox_interfaces.rb]: Yes, created another PR for it

File: app/helpers/proxmox_vm_attrs_helper.rb (hunk 3 above)
Comment (Manisha15) [reply app/helpers/proxmox_vm_attrs_helper.rb:121]: Some attributes are part of config_attributes hash within vm_attrs hash and some are direct vm_attributes hash. This loop assign names to them accordingly.

**Pull request merged**